Src/CZICmd/cmdlineoptions.cpp

```diff
@@ -2279,3 +2313,76 @@ void CCmdLineOptions::PrintHelpStreamsObjects()
 
     return true;
 }
+
+/*static*/bool CCmdLineOptions::TryParseSubBlockCacheSize(const std::string& text, std::uint64_t* size)
+{
+    regex regex(R"(^\s*([+]?(?:[0-9]+(?:[.][0-9]*)?|[.][0-9]+))\s*(k|m|g|t|ki|mi|gi|ti)(?:b?)\s*$)", regex_constants::icase);
```

Comment:
- Is the start of line + whitespace matching necessary?
  - I am assuming this is to avoid having to trim whitespace from incoming text
- Why is + conditionally matched for cache size?
- Could you split up the regex to document what is actually occurring?. For Example
```
R"^\s*" // Match any whitespace at beginning of line
R"("    // First Capture Group
R"[+]?" // Match + if it exists before out number (why?) 
... etc
```

My understanding is that neighboring literal strings will be composed and this allows for more explicit commenting rather than relying on reader to have a good enough understanding of regex to entirely understand intent in the intricacies of string

Reply:
I have to confess - I'd recommend pasting the regex into e.g. https://regex101.com/ if one ever needs to understand/modify it.
I asked Copilot for help with documenting the regex and added this.